gflow/gflow.py

```diff
@@ -1,107 +1,237 @@
-import sys
 import json
 import logging
-from input import Input
-from config import Config
+import yaml
+import os
 from bioblend.galaxy.objects import *
-__version__ = "0.1.0"
 
 
 class GFlow(object):
-
-    def __init__(self, config):
-
+    def __init__(self, configfile=None, galaxy_url=None, galaxy_key=None, library_name=None, history_name=None,
+                 workflow_source=None, workflow=None, datasets_source=None, datasets=None, runtime_params=None):
+        """
+        Interact with a Galaxy instance.
+
+        Args:
+            configfile (str): The location of a config file.
+        Attributes:
+            self.logger: For logging.
+            self.cfg (dict): A dictionary to hold all the parameters provided by the config argument.
+            self.galaxy_url (str): The URL of an instance of Galaxy
+            self.galaxy_key (str): The API key of an instance of Galaxy
+        """
         self.logger = logging.getLogger('gflow.GFlow')
-
-        self.options = config
-        self.datasets = []
-        for i in range(0, self.options.num_datasets):
-            self.datasets.append(Input(self.options.dataset_src, self.options.datasets[i], self.options.labels[i]))
+        if configfile:
+            self.logger.info("Reading configuration file")
+            with open(configfile, "r") as ymlfile:
+                config = yaml.load(ymlfile)
+            self.logger.info("Checking config file for empty values")
+            self.check_for_empty_values(config)
+            try:
+                self.galaxy_url = config['galaxy_url']
+                self.galaxy_key = config['galaxy_key']
+                self.library_name = config['library']
+                self.history_name = config['history']
+                self.workflow_source = config['workflow_source']
+                self.workflow = config['workflow']
+            except KeyError:
+                self.logger.error("Missing required parameter(s)")
```

Comment:
Consider adding the keys that are required in the error message.  Otherwise, how does the user know what they are.
Also, when raising an exception, considering adding a message:
    try:
      raise ValueError;  # something bad...
    except ValueError as err:
      err.message=err.message+" hello"
      raise